### DISCUSSION TITLE
Loading texture from tiff images

### DISCUSSION THREAD

**Discussion by edmBernard** [bevyengine/bevy#6686] [Q&A] [answered]
I try to load texture from tiff image. 
and I got the following error : 
```no `AssetLoader` found for the following extension: tiff```
After some search I found that bevy have a [ImageTextureLoader](https://docs.rs/bevy/0.9.0/bevy/render/texture/struct.ImageTextureLoader.html#) that should support all format supported by the crates `image`.
But I have no clue how to use it.
I also found that we can create bevy Image from DynamicImage (image-rs format). 
I haven't found how to put the texture on a sprite without the asset-server.
Does someone have an idea ?

**Comment by DasLixou** [comment]
You have to add the `tiff` feature to bevy

**Reply by edmBernard** [reply]
There is no `tiff` feature in bevy : https://github.com/bevyengine/bevy/blob/HEAD/docs/cargo_features.md

**Reply by edmBernard** [reply]
I try to enable all features, it seem that if I enable bevy_render I can read "tiff" image not "tif" and only tiff in 8bit RGB.
We can't read tiff in 16UInt and tiff 8bit Palette (the 8b Palette is not supported by the crates `image` so I don't expect it's support).

**Reply by DasLixou** [reply]
Then there is probably no current way to enable it. You could make a pr introducing it as a feature when you are sure that there would be no problem in the bevy code and the image library could handle it fine

**Reply by edmBernard** [reply]
I found a way to make it work, It's manual. 
I use the method shown here https://www.reddit.com/r/bevy/comments/y5fzkl/how_to_get_a_texture_from_a_in_memory_image_not_a/
I manually load the image with image-rs and send the handle to the texture.

**Reply by edmBernard** [reply]
the issue with 16U is the wgpu doesn't seem to support this format for texture. 
with the manual load I can convert 16u image to 8u on the loading code.
thanks for the help

**Comment by edmBernard** [comment, marked as answer]
A hacky solution : 
https://github.com/bevyengine/bevy/discussions/6686#discussioncomment-4183455
> I found a way to make it work, It's manual. I use the method shown here https://www.reddit.com/r/bevy/comments/y5fzkl/how_to_get_a_texture_from_a_in_memory_image_not_a/ I manually load the image with image-rs and send the handle to the texture.
